addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -146,12 +146,14 @@ def write(self, vals):
             if batch_without_picking_type:
                 picking = self.picking_ids and self.picking_ids[0]
                 batch_without_picking_type.picking_type_id = picking.picking_type_id.id
+        if vals.get('state') and vals.get('state') == 'cancel':
```

Comment:
If action_cancel is the only path to set the state to `cancel`. I would suggest setting this logic inside it. It's maybe cleaner and it's allow you to make only one call to write (instead of 2 here with the `self.picking_ids = False`)

Reply:
[It's not!](https://github.com/odoo/odoo/blob/master/addons/stock_picking_batch/models/stock_picking.py#L27). It was merged after I did the work (https://github.com/odoo/odoo/pull/66600) and I was too lazy to rebase since there wasn't a conflict.

Reply:
Actually never mind, I'm a dummy and you are right. Made the change.